maas-api.yaml

```diff
@@ -1,317 +1,383 @@
                 modes:
-                  - mode: pt_pub_bus
-                    numberOfServices: 7702
-                    realtime:
-                      - alerts: true
-                      - positions: true
-                      - updates: false
+                  - pt_pub_tram
+                  - pt_pub_bus
+                full: true
+      responses:
+        200:
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: object
+                  description: TSP information.
+                  properties:
+                    id:
+                      type: string
+                      description: TSP ID.
+                    name:
+                      type: string
+                      description: TSP name.
+                    modes:
+                      type: array
+                      items:
+                        description: Detailed information for this agency under a specific public transport mode.
+                        type: object
+                        properties:
+                          mode:
+                            type: string
+                            description: Agency's mode identifier.
+                          numberOfServices:
+                            type: integer
+                            description: Total number of services on the timetable this agency is running within a typical week.
+                          realTime:
+                            $ref: '#/components/schemas/RealTimeData'
+                example:
+                  - id: LACMTA
+                    name: Metro - Los Angeles
+                    modes:
+                      - mode: pt_pub_tram
+                        numberOfServices: 5698
+                      - mode: pt_pub_bus
+                        numberOfServices: 36855
+                        realTime:
+                          - alerts: false
+                          - positions: true
+                          - updates: true
+                  - id: 27
+                    name: LADOT
+                    modes:
+                      - mode: pt_pub_bus
+                        numberOfServices: 15828
+                  - id: 6216179
+                    name: Big Blue Bus
+                    modes:
+                      - mode: pt_pub_bus
+                        numberOfServices: 7702
+                        realTime:
+                          - alerts: true
+                          - positions: true
+                          - updates: false
         400:
           description: Bad request, most likely due to issues with the input or non-posible scenarios.
-          schema:
-            $ref: '#/definitions/InputError'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InputError'
 
   /info/route.json:
```

Comment:
Hmm, good question. An issue I see with this is that regions could overlap, which makes this (and the other) endpoints harder to use. How about making agency IDs globally unique, and just use `/agencies/{agencyID}/routes`? (A stop gap measure to make them globally unique is to concatenate region ID + agency ID to make it a agency GUID, though you could get the same problem of unexpected behaviour with overlapping regions.)

Reply:
Though this would mean that IDs don't match GTFS data anymore, but I think this is fine for a transport aggregator API like this. Possibly best to make this explicit by having a `gtfsID` for everything and naming the IDs of this API ` GUID`.